File: app/src/main/java/ru/mamykin/foreignbooksreader/ui/activities/BookDetailsActivity.java
package ru.mamykin.foreignbooksreader.ui.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.widget.TextView;

import com.arellomobile.mvp.presenter.InjectPresenter;
import com.arellomobile.mvp.presenter.ProvidePresenter;

import butterknife.BindView;
import butterknife.OnClick;
import ru.mamykin.foreignbooksreader.R;
import ru.mamykin.foreignbooksreader.common.UiUtils;
import ru.mamykin.foreignbooksreader.presenters.BookDetailsPresenter;
import ru.mamykin.foreignbooksreader.views.BookDetailsView;

/**
 * Страница с информацией о книге
 */
public class BookDetailsActivity extends BaseActivity implements BookDetailsView {
    private static final String BOOK_ID_EXTRA = "book_id_extra";

    @BindView(R.id.tvName)
    protected TextView tvBookName;
    @BindView(R.id.fabRead)
    protected FloatingActionButton fabRead;
    @BindView(R.id.tvAuthor)
    protected TextView tvBookAuthor;
    @BindView(R.id.tvBookPath)
    protected TextView tvBookPath;
    @BindView(R.id.tvBookCurrentPage)
    protected TextView tvBookCurrentPage;
    @BindView(R.id.tvBookGenre)
    protected TextView tvBookGenre;
    @BindView(R.id.tvBookLanguage)
    protected TextView tvBookLanguage;
    @BindView(R.id.tvBookCreatedDate)
    protected TextView tvBookCreatedDate;

    @InjectPresenter
    BookDetailsPresenter presenter;

    @ProvidePresenter
    BookDetailsPresenter provideBookDetailsPresenter() {
        return new BookDetailsPresenter(getIntent().getExtras().getInt(BOOK_ID_EXTRA));
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initToolbar(getString(R.string.about_book), true);
    }

    @Override
    public int getLayout() {
        return R.layout.activity_book_detail;
    }

    @Override
    public void showTitle(String title) {
        UiUtils.setTitle(this, title);
    }

    @Override
    public void setHomeEnabled(boolean enabled) {
        UiUtils.setHomeEnabled(this, enabled);
    }

    @Override
    public void showBookName(String name) {
        tvBookName.setText(name);
    }

    @Override
    public void showBookAuthor(String author) {
        tvBookAuthor.setText(author);
    }

    @Override
    public void showBookPath(String path) {
        tvBookPath.setText(path);
    }

    @Override
    public void showBookCurrentPage(String currentPage) {
        tvBookCurrentPage.setText(currentPage);
    }

    @Override
    public void showBookGenre(String genre) {
        tvBookGenre.setText(genre);
    }

    @Override
    public void showBookOriginalLang(String lang) {
        tvBookLanguage.setText(lang);
    }

    @Override
    public void showBookCreatedDate(String date) {
        tvBookCreatedDate.setText(date);
    }

    @Override
    public void openBook(int bookId) {
        startActivity(ReadBookActivity.getStartIntent(this, bookId));
    }

    @OnClick(R.id.fabRead)
    public void onReadClicked() {
        presenter.onReadClicked();
    }

    public static Intent getStartIntent(Context context, int bookId) {
        final Intent bookDetailsIntent = new Intent(context, BookDetailsActivity.class);
        bookDetailsIntent.putExtra(BookDetailsActivity.BOOK_ID_EXTRA, bookId);
        return bookDetailsIntent;
    }
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/ui/activities/BookDetailsActivity.kt
package ru.mamykin.foreignbooksreader.ui.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.widget.TextView

import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.ProvidePresenter

import butterknife.BindView
import butterknife.OnClick
import ru.mamykin.foreignbooksreader.R
import ru.mamykin.foreignbooksreader.common.UiUtils
import ru.mamykin.foreignbooksreader.presenters.BookDetailsPresenter
import ru.mamykin.foreignbooksreader.views.BookDetailsView

/**
 * Страница с информацией о книге
 */
class BookDetailsActivity : BaseActivity(), BookDetailsView {

    @BindView(R.id.tvName)
    protected var tvBookName: TextView? = null
    @BindView(R.id.fabRead)
    protected var fabRead: FloatingActionButton? = null
    @BindView(R.id.tvAuthor)
    protected var tvBookAuthor: TextView? = null
    @BindView(R.id.tvBookPath)
    protected var tvBookPath: TextView? = null
    @BindView(R.id.tvBookCurrentPage)
    protected var tvBookCurrentPage: TextView? = null
    @BindView(R.id.tvBookGenre)
    protected var tvBookGenre: TextView? = null
    @BindView(R.id.tvBookLanguage)
    protected var tvBookLanguage: TextView? = null
    @BindView(R.id.tvBookCreatedDate)
    protected var tvBookCreatedDate: TextView? = null

    @InjectPresenter
    internal var presenter: BookDetailsPresenter? = null

    @ProvidePresenter
    internal fun provideBookDetailsPresenter(): BookDetailsPresenter {
        return BookDetailsPresenter(intent.extras!!.getInt(BOOK_ID_EXTRA))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initToolbar(getString(R.string.about_book), true)
    }

    override fun getLayout(): Int {
        return R.layout.activity_book_detail
    }

    override fun showTitle(title: String) {
        UiUtils.setTitle(this, title)
    }

    override fun setHomeEnabled(enabled: Boolean) {
        UiUtils.setHomeEnabled(this, enabled)
    }

    override fun showBookName(name: String) {
        tvBookName!!.text = name
    }

    override fun showBookAuthor(author: String) {
        tvBookAuthor!!.text = author
    }

    override fun showBookPath(path: String) {
        tvBookPath!!.text = path
    }

    override fun showBookCurrentPage(currentPage: String) {
        tvBookCurrentPage!!.text = currentPage
    }

    override fun showBookGenre(genre: String) {
        tvBookGenre!!.text = genre
    }

    override fun showBookOriginalLang(lang: String) {
        tvBookLanguage!!.text = lang
    }

    override fun showBookCreatedDate(date: String) {
        tvBookCreatedDate!!.text = date
    }

    override fun openBook(bookId: Int) {
        startActivity(ReadBookActivity.getStartIntent(this, bookId))
    }

    @OnClick(R.id.fabRead)
    fun onReadClicked() {
        presenter!!.onReadClicked()
    }

    companion object {
        private val BOOK_ID_EXTRA = "book_id_extra"

        fun getStartIntent(context: Context, bookId: Int): Intent {
            val bookDetailsIntent = Intent(context, BookDetailsActivity::class.java)
            bookDetailsIntent.putExtra(BookDetailsActivity.BOOK_ID_EXTRA, bookId)
            return bookDetailsIntent
        }
    }
}